src/shadowbox/infrastructure/logging.ts

```diff
@@ -12,25 +12,45 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import * as path from 'path';
 
-function makeLogMessage(level: string, message: string): string {
+interface Callsite {
+  getLineNumber(): number;
+  getFileName(): string;
+}
+
+function getCallsite(): Callsite {
+  const originalPrepareStackTrace = Error.prepareStackTrace;
+  Error.prepareStackTrace = (_, stack) => {
+      return stack;
+  };
+  const error = new Error();
+  Error.captureStackTrace(error, getCallsite);
+  // tslint:disable-next-line:no-any
+  const stack = error.stack as any as Callsite[];
+  Error.prepareStackTrace = originalPrepareStackTrace;
+  return stack[1];
+}
+
+function makeLogMessage(level: string, callsite: Callsite, message: string): string {
```

Comment:
Suggestion: define the three possible values for `level`

Reply:
Done